### PR TITLE
[OpenVINO] Enable MiniCPM-V-4.5 3D-Resampler temporal grouping

### PR DESCRIPTION
**Pull request by BiswarupMukherjee1**
## What does this PR do?

MiniCPM-V-4.5's 3D-Resampler jointly compresses several consecutive video frames into a single
`query_num`-token group. The OpenVINO path never consumed `temporal_ids`: `resampling()` did not
take them, and the resampler sub-model is driven with three per-frame inputs. So every frame
produced its own 64-token group, MiniCPM-V-2.6 behaviour, even though the checkpoint config
sets `batch_3d_resampler: true`.

Upstream performs the fold inside `Resampler.batch_attn_forward` using
`torch.nn.utils.rnn.pad_sequence`, which the OpenVINO PyTorch frontend cannot convert
(`aten::pad_sequence` has no translator and is listed under "Do not work on these for now" in
openvinotoolkit/openvino#28584).

**This PR does the fold on the host instead, and changes nothing in the graph.** `kv_proj`
(Linear) and `ln_kv` (LayerNorm over the last dim) act per token, so they commute with
regrouping; the per-frame temporal vector is therefore folded into the per-token `pos_embed`
that the existing graph already adds to `k`. Consequently:

* no new resampler IR input,
* no exporter or model-patcher change,
* no `aten::pad_sequence` support required,
* **already-exported resampler IRs run the grouped path unchanged.**

Temporal position embeddings are computed for the ids actually present rather than cached for
`max_temporal_size` (72000) rows, which at `hidden_size = 4096` would reserve ~1.2 GB for a
handful of rows that are ever indexed.

`temporal_ids=None` reproduces the previous per-frame behaviour bit-for-bit, so MiniCPM-V-2.6
and every image-only path are unaffected.

## Verification

Checked against the **unmodified upstream `Resampler`** with `batch_infer=True` (the path the
real checkpoint takes):

| case | result |
|---|---|
| uniform packing=2 (8 frames → 4 groups) | cosine 1.000000 |
| ragged packing=3 (8 frames → 3 groups) | cosine 1.000000 |
| packing=6 (6 frames → 1 group) | cosine 1.000000 |
| ragged `tgt_sizes` within a group | cosine 1.000000 |
| `temporal_ids=None` | **bit-identical** to current behaviour (`max\|diff\| == 0.0`) |

Also verified end to end through an OpenVINO IR: the three-input resampler graph was converted
once with dynamic shapes, then packings 1/2/3/4/6 were each driven through that single compiled
model, matching the reference every time.

The added unit test (`test_minicpmv_3d_resampler_temporal_grouping`) needs no model download —
it drives `resampling()` with a stub resampler and asserts the group count, the frame-major
fold, the ragged padding/mask, the "grouping must cover every frame" guard, and the
`temporal_ids=None` no-regression case.

I kept the test dependency-free deliberately, but `optimum-intel-internal-testing/tiny-random-minicpm-v-4_5`
already exists in the testing org (`version: 4.5`, `batch_3d_resampler: true`, `query_num: 64`)
and is not yet wired into `MODEL_NAMES`. Happy to add an end-to-end export/generate test on top
of it if you would prefer that, either here or as a follow-up.

## End-to-end demonstration

@rkazants  - here is a self-contained script showing exactly the case this fixes. It uses
MiniCPM-V-4.5's **own** `Resampler` as the reference, and downloads a single ~20 kB source file;
no model weights are required.

```python
# Demonstrates the case fixed by this PR, using MiniCPM-V-4.5's OWN Resampler as the
# reference. Downloads a single ~20 kB source file; no model weights required.
import importlib.util, types, torch, numpy as np
from huggingface_hub import hf_hub_download
from optimum.intel.openvino.modeling_visual_language import _OVMiniCPMVForCausalLM

# --- MiniCPM-V-4.5's reference Resampler (the behaviour OpenVINO should reproduce) ---
spec = importlib.util.spec_from_file_location(
    "mcpm_resampler", hf_hub_download("openbmb/MiniCPM-V-4_5", "resampler.py"))
mcpm = importlib.util.module_from_spec(spec); spec.loader.exec_module(mcpm)

EMBED, KV, QUERIES = 128, 64, 64          # query_num=64 as in the 4.5 config
torch.manual_seed(0)
ref = mcpm.Resampler(num_queries=QUERIES, embed_dim=EMBED, num_heads=4, kv_dim=KV,
                     adaptive=False, max_size=(8, 8), max_temporal_size=64,
                     batch_infer=True).eval()          # batch_3d_resampler: true

# --- a 6-frame clip whose processor grouped frames into 2 temporal groups of 3 ---
FRAMES, temporal_ids = 6, [[0, 1, 2], [3, 4, 5]]
tgt_sizes = torch.tensor([[2, 3]] * FRAMES, dtype=torch.int32)
torch.manual_seed(1)
vision_out = torch.randn(FRAMES, 6, KV)               # vision-encoder output

with torch.no_grad():
    expected = ref(vision_out, tgt_sizes=tgt_sizes, temporal_ids=temporal_ids)

# --- drive optimum-intel's resampling() with a stand-in for the exported resampler IR ---
class _Model:
    resampling = _OVMiniCPMVForCausalLM.resampling
    _temporal_pos_embed_rows = _OVMiniCPMVForCausalLM._temporal_pos_embed_rows
    embed_dim, _pos_embeds = EMBED, ref.pos_embed.float()
    def _adjust_pos_cache(self, tgt_sizes): pass
    @torch.no_grad()
    def resampler(self, image_feature, pos_embed, key_padding_mask):   # == OVResamplerModel
        x = ref.ln_kv(ref.kv_proj(image_feature)).permute(1, 0, 2)
        q = ref.ln_q(ref.query).unsqueeze(1).repeat(1, x.shape[1], 1)
        o = ref.attn(q, x + pos_embed, x, key_padding_mask=key_padding_mask)[0]
        return (ref.ln_post(o.permute(1, 0, 2)) @ ref.proj).numpy()

m = _Model()
before = m.resampling(vision_out, tgt_sizes)                              # today
after  = m.resampling(vision_out, tgt_sizes, temporal_ids=temporal_ids)   # with this PR

cos = lambda a, b: float(np.dot(np.ravel(a), np.ravel(b)) /
                         (np.linalg.norm(a) * np.linalg.norm(b)))
print(f"frames={FRAMES}  temporal groups={len(temporal_ids)}  query_num={QUERIES}\n")
print(f"MiniCPM-V-4.5 reference : {tuple(expected.shape)}  -> {expected.shape[0]*QUERIES} vision tokens")
print(f"optimum-intel (before)  : {before.shape}  -> {before.shape[0]*QUERIES} vision tokens   <-- one group PER FRAME")
print(f"optimum-intel (after)   : {after.shape}  -> {after.shape[0]*QUERIES} vision tokens")
print(f"\ncosine(after, reference) = {cos(after, expected.numpy()):.6f}")
print(f"max|after - reference|   = {float(np.abs(np.asarray(after) - expected.numpy()).max()):.2e}")
```

Output:

```
frames=6  temporal groups=2  query_num=64

MiniCPM-V-4.5 reference : (2, 64, 128)  -> 128 vision tokens
optimum-intel (before)  : torch.Size([6, 64, 128])  -> 384 vision tokens   <-- one group PER FRAME
optimum-intel (after)   : torch.Size([2, 64, 128])  -> 128 vision tokens

cosine(after, reference) = 1.000000
max|after - reference|   = 1.67e-06
```

**The case fixed:** a 6-frame clip whose processor grouped the frames into 2 temporal groups.
MiniCPM-V-4.5's reference implementation emits **one 64-token group per temporal group** (2
groups -> 128 vision tokens). Before this PR, optimum-intel emitted **one group per frame** (6
-> 384 tokens) because `resampling()` had no way to receive `temporal_ids`. After it, the output
matches the reference exactly.

Beyond the token count, the mismatch is a correctness problem: the number of resampler rows must
equal the number of image placeholder slots the processor put in the prompt, so a grouped video
input produces a different embedding layout than the model was trained for.

## Rebase status

Rebased and re-verified against `main` @ `2039025` (2026-09-08). No conflicts; the new test and
`black`/`ruff` (pinned `black~=23.1`, `ruff==0.4.4`) all pass on the rebased tree. None of the commits merged since this PR opened touch `resampling()`, `get_vision_embeddings()` or any
temporal handling.

## Relationship to the other open MiniCPM PRs

* **#1941** adds `temporal_ids` to `forward()` / `prepare_inputs_for_generation()` so
  `generate()` stops raising `ValueError("model_kwargs not used: ['temporal_ids']")`. Its test
  notes the runtime should "accept and safely ignore" them — correct for still images (all ids
  are `-1`, contributing a zero embedding), but it means grouped **video** is silently
  discarded. This PR is the other half: #1941 makes the argument arrive, this makes it mean
  something. They compose; happy to rebase if #1941 lands first.
* **#1412** adds MiniCPM-V-4/4.5 decoder plumbing plus a 4-input resampler carrying
  `temporal_embed`, doing one attention call with `bs = n_frames`. That applies the temporal
  *embedding* but not the temporal *fold*, so it still emits one vector per frame where upstream
  emits one per group. Folding into `pos_embed` makes the fourth input unnecessary. Happy to
  rebase onto that branch instead if the maintainers prefer.
* **#1945 / #1906 / #1810** target MiniCPM-V-4.6, which replaced the resampler with a spatial
  `merge_kernel_size` merger and has no temporal path, so they do not cover this.

## Before submitting
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests)?
- [x] Did you write any new necessary tests?

### CODE REVIEW

**Review by rkazants — COMMENTED** [review]
please provide e2e code snippet in PR description to demo what case is exactly fixed.

**Comment by BiswarupMukherjee1** [comment]
> please provide e2e code snippet in PR description to demo what case is exactly fixed.

Thanks @rkazants , added an end-to-end snippet to the PR description. It uses MiniCPM-V-4.5's own Resampler as the reference and needs no model weights (~20 kB source download), so it runs as-is.

Summary: for a 6-frame clip grouped into 2 temporal groups, the reference emits 2 × 64 = 128 vision tokens, while optimum-intel currently emits 6 × 64 = 384, one group per frame instead of per temporal group, because resampling() had no way to receive temporal_ids. With this PR the output matches the reference at cosine 1.000000.

I've also rebased on current main (2039025) and re-run the test and style checks, no conflicts.
